### ISSUE TITLE
[BUG] MessageLockLostException in Microsoft.Azure.ServiceBus

### ISSUE THREAD

**Issue by allpay-raj** [Azure/azure-sdk-for-net#38295]
### Library name and version

Microsoft.Azure.ServiceBus 4.2.1

### Describe the bug

I am deploying function app with service bus trigger. The function app has multiple functions listening with different subscriptions to same topic. Before deploying the function app to production slot, we disable the subscription using az command as the message needs to be processed only once as it does registration of user. 

`az servicebus topic subscription update --resource-group xxx --namespace-name yyy --topic-name mytopic --name subscription --status "ReceiveDisabled"`

After the app is deployed we enable the subscription

`az servicebus topic subscription update --resource-group xxx  --namespace-name yyy  --topic-name mytopic  --name subscription  --status "Active"`

### Expected behavior

We expect all the message in the queue to be processed 

### Actual behavior

Roughly half the message throw following error message when we are trying to Complete the message

**Exception while executing function: CartaDispatcherSendRegistration The lock supplied is invalid. Either the lock expired, or the message has already been removed from the queue, or was received by a different receiver** 

The subscription is set to following
lockout dution : 1m 15 mins
AutoComplete: Off
maxAutoRenewDuration 5 mins
maxConcurrentCalls: 16 (default)
Max delivery count: 1

Now what is interesting is this error only occurs when I disable and enable Subscription. If I stop the function and let the queue build up and then  enable the function, there is no error. 

If I reduce maxConcurrentCalls down to 1, then it works fine with no errors but this has a significant impact on performance

### Reproduction Steps

1. Create a very simple app listening to a topic and subscription
 `using System;
using System.Threading.Tasks;
using Microsoft.Azure.ServiceBus.Core;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Host;
using Microsoft.Extensions.Logging;

namespace ServiceBus
{
    public class Function1
    {
        private readonly ILogger<Function1> _logger;

        public Function1(ILogger<Function1> log)
        {
            _logger = log;
        }

        [FunctionName("Function1")]
        public async Task Run([ServiceBusTrigger("partition-topic", "hello-world", Connection  = "AzureServiceBus")]string mySbMsg, string lockToken, MessageReceiver messageReceiver,  ILogger logger)
        {
            messageReceiver.PrefetchCount = 0;
            messageReceiver.
            _logger.LogInformation($"C# ServiceBus topic trigger function processed message: {mySbMsg}");
            Console.WriteLine(mySbMsg);
            try
            {
                await messageReceiver.CompleteAsync(lockToken);
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Error: {ex}");
                await messageReceiver.DeadLetterAsync(lockToken);
            }

        }
    }
}
`
2. Start the function app
3. Set the Subscription to ReceiveDisabled with use az or portal
4. Send some 100 test messages
5. Set the Subscription to Enable
6. See the app throw errors

I am using .Net 6

### Environment

.NET SDK:
 Version:   7.0.306
 Commit:    f500069cb7

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19045
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.306\

Host:
  Version:      7.0.9
  Architecture: x64
  Commit:       8e9a17b221
.NET SDKs installed:
  3.1.426 [C:\Program Files\dotnet\sdk]
  5.0.408 [C:\Program Files\dotnet\sdk]
  5.0.414 [C:\Program Files\dotnet\sdk]
  6.0.120 [C:\Program Files\dotnet\sdk]
  6.0.315 [C:\Program Files\dotnet\sdk]
  7.0.306 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.25 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.25 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.20 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.25 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

**Comment by allpay-raj** [comment]
Tried same thing with Azure.Messaging.ServiceBus 5.1.2 and there is no error. Works fine as expected. 

Here is the code

`using Azure.Messaging.ServiceBus;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.ServiceBus;
using Microsoft.Extensions.Logging;
using System.Threading.Tasks;

namespace ServiceBus
{
    public class Function1
    {
        private readonly ILogger<Function1> _logger;

        public Function1(ILogger<Function1> log)
        {
            _logger = log;
        }

        [FunctionName("Function1")]
        public async Task Run([ServiceBusTrigger("partition-topic",  "hello-world", Connection  = "AzureServiceBus")] ServiceBusReceivedMessage msg,  ServiceBusMessageActions messageActions,  ILogger log)
        {
            log.LogInformation($"C# ServiceBus queue trigger function processed message: {msg.Body}");

            await messageActions.CompleteMessageAsync(msg);
            //log.LogInformation($"EnqueuedTimeUtc={enqueuedTimeUtc}");
            //log.LogInformation($"DeliveryCount={deliveryCount}");
            //log.LogInformation($"MessageId={messageId}");

        }
    }
}`

host.json

`{
  "version": "2.0",
  "logging": {
    "applicationInsights": {
      "samplingSettings": {
        "isEnabled": true,
        "excludedTypes": "Request"
      },
      "enableLiveMetricsFilters": true
    }
  },
  "extensions": {
    "serviceBus": {
      "clientRetryOptions": {
        "mode": "exponential",
        "tryTimeout": "00:01:00",
        "delay": "00:00:00.80",
        "maxDelay": "00:01:00",
        "maxRetries": 3
      },
      "prefetchCount": 0,
      "transportType": "amqpWebSockets",
      "autoCompleteMessages": false,
      "maxAutoLockRenewalDuration": "00:05:00",
      "maxConcurrentCalls": 16,
      "maxConcurrentSessions": 8,
      "maxMessageBatchSize": 1000,
      "minMessageBatchSize": 1,
      "maxBatchWaitTime": "00:00:30",
      "sessionIdleTimeout": "00:01:00",
      "enableCrossEntityTransactions": false
    }
  }
}`

**Comment by jsquire** [comment]
@allpay-raj :  Please provide the version of the `Microsoft.Azure.WebJobs.Extensions.ServiceBus` package that you're using.   Also, please be aware that `Microsoft.Azure.ServiceBus` is deprecated and receiving only critical bug and security fixes.   We recommend using `Azure.Messaging.ServiceBus` for any new development.

**Comment by allpay-raj** [comment]
Thanks @jsquire we are using v 4.3.1 for `Microsoft.Azure.WebJobs.Extensions.ServiceBus`

**Comment by jsquire** [comment]
Hi @allpay-raj.  Unfortunately, the packages that you're using are using versions that are well out of date.   `Microsoft.Azure.ServiceBus` v4.2.1 is more than 2.5 years old and multiple versions behind; as mentioned, the library itself is deprecated and this scenario would not meet the bar for a critical bug fix.  Likewise, the 4.x line of `Microsoft.Azure.WebJobs.Extensions.ServiceBus` is no longer under active maintenance and has been superseded by the 5.x line.

We'd strongly recommend upgrading to the latest `Microsoft.Azure.WebJobs.Extensions.ServiceBus` and `Azure.Messaging.ServiceBus` packages.  If you are, for some reason, unable to do so - you may try bumping `Microsoft.Azure.ServiceBus` to [v5.2.0](https://www.nuget.org/packages/Microsoft.Azure.ServiceBus/5.2.0) which is the latest release - although it is also nearly two years old.  As a last-ditch effort, you may want to try taking a direct reference to [Microsoft.Azure.Amqp v2.6.3](https://www.nuget.org/packages/Microsoft.Azure.Amqp/2.6.3) in your project, which will hoist the version used by Service Bus and may help.

I'm going to mark this as addressed, but please feel free to unresolve if you'd like to continue the discussion.

**Comment by allpay-raj** [comment]
Thanks @jsquire. We did try bumping 'Azure.Messaging.ServiceBus' to 5.2.0 but still the same error. I will try bumping [Microsoft.Azure.Amqp](https://www.nuget.org/packages/Microsoft.Azure.Amqp/2.6.3) and see if it solves the issue

**Comment by allpay-raj** [comment]
@jsquire tried  bumping `Microsoft.Azure.ServiceBus` to 5.2.0 and added [Microsoft.Azure.Amqp](https://www.nuget.org/packages/Microsoft.Azure.Amqp/2.6.3) - still the same error. I guess we will have to bite the bullet and upgrade to `Azure.Messaging.ServiceBus`. Thanks for your support
